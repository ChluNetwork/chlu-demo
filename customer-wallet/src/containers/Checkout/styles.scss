.checkout {

   &-seller {

     &__header {
       display: flex;
       align-items: center;

       &-name {
         margin-right: 60px;
         color: rgb(0, 188, 212);
         font-size: 20px;
       }

       &-info {
         font-size: 12px;
         color: #808080;
       }
     }
     
     &__product {
       display: flex;
       margin-top: 30px;

       > img {
         max-height: 150px;
       }
       
       &-description {
         margin-left: 30px;
       }
     }

     &__pay-buttons {
       max-width: 300px;
     }

     &__pay-button {
       margin: 10px;

       &:first-child, &:last-child {
         margin-left: 0;
       }
     }
   }
}
