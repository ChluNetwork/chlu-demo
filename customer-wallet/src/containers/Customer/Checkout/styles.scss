@import 'styles/variables.scss';

.checkout {
  .checkout-vendor {

    .section-content {
      border: 1px solid $borderColor;
      border-radius: 3px;
    }

    .product {

      &-info {
        padding: 10px;
        border-bottom: 1px solid $borderColor;

        &__name {
          font-weight: bold;
        }

        &__price {
          font-weight: 100;
          margin-top: 5px;
        }

        &__rating {
          font-size: 25px;
        }
      }

      &-img{
        padding: 15px 10px;
        display: flex;
        justify-content: center;

        &__wrapper {
          max-width: 200px;

          .img {
            width: 100%;
          }
        }
      }
    }

    .payment-label {
      margin: 20px 0 10px;
      font-weight: bold;
    }

    .payment {
      &-item {}
    }

    &__button {
      padding: 15px 0;
    }
  }
}
