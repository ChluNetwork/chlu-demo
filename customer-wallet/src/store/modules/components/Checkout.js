import { handleActions } from 'redux-actions'

// ------------------------------------
// Constants
// ------------------------------------
const initialState = {
  rating: 4
}

// ------------------------------------
// Reducer
// ------------------------------------
export default handleActions({}, initialState)
