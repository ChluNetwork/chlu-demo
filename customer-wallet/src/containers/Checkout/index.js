import React from 'react'
// components
import HeaderSection from './sections/HeaderSection'
import PaySection from './sections/PaySection'
import ProductSection from './sections/ProductSection'
// styles
import './styles.css'

const Checkout = () => (
  <div className='page-container checkout'>
    <h3>Checkout at Etsy</h3>
    <div className='checkout-seller'>
      <HeaderSection />
      <span className='checkout-seller__header-info'>
        (250 Reviews, Over $250,000 sales)
      </span>
      <ProductSection />
      <PaySection />
    </div>
  </div>
)

export default Checkout
