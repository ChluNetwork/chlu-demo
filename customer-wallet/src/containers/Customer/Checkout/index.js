import React, { Component } from 'react'
import PropTypes from 'prop-types'
// redux
import { connect } from 'react-redux'
import { actions } from 'shared-libraries/lib'
// components
import RaisedButton from 'material-ui/RaisedButton'
import PaySection from './sections/PaySection'
import ProductSection from './sections/ProductSection'
import CircularProgress from 'material-ui/CircularProgress'
// styles
import './styles.css'
// constants
import { orange } from 'context/palette'

const { dataActions: { checkout: { getCheckout } } } = actions

const buttonStyles = { border: '1px solid rgb(0, 0, 0)' }

const labelStyle = { textTransform: 'none' }

class Checkout extends Component {
  componentDidMount() {
    const { getCheckout } = this.props

    getCheckout()
  }

  render() {
    const { checkout: {
      loading,
      data: { rating, avatar, price, product }
    } } = this.props

    return (
      <div className='page-container checkout color-main'>
        <div className='container'>
          {
            loading
              ? <CircularProgress />
              : <div className='checkout-vendor'>
                  <ProductSection product={product} price={price} avatar={avatar} rating={rating} />
                  <div className='payment-label'>Payment Method</div>
                  <PaySection />
                  <div className='checkout-vendor__button'>
                    <RaisedButton
                      label='Continue'
                      fullWidth={true}
                      style={buttonStyles}
                      backgroundColor={orange}
                      labelStyle={labelStyle}
                      onClick={() => null}
                    />
                  </div>
              </div>
          }
        </div>
      </div>
    )
  }
}

Checkout.propTypes = {
  checkout: PropTypes.object.isRequired,
  getCheckout: PropTypes.func.isRequired
}

const maoStateToProps = store => ({
  checkout: store.data.checkout
})

const mapDispatchToProps = {
  getCheckout
}

export default connect(maoStateToProps, mapDispatchToProps)(Checkout)
