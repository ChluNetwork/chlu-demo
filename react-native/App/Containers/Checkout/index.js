import React, { Component } from 'react'
// rn-components
import { View, Text, StatusBar, ScrollView } from 'react-native'
// other components
import DrawerButton from '../../Components/DrawerButton'
import Product from './components/Product'
import PaymentTypes from './components/PaymentTypes'
import ContinueButton from './components/ContinueButton'
// styles
import styles from '../Styles/CheckoutStyles'

const options = [
  { label: 'Chlu', value: 0 },
  { label: 'Credit card', value: 1 },
  { label: 'PayPal', value: 2 }
]

class Checkout extends Component {
  static navigationOptions = ({ navigation }) => ({
    headerLeft: <DrawerButton action={() => navigation.navigate('DrawerOpen')} />
  })

  state = {
    selectedPayment: 0
  }

  handleSelectChange = (value) => {
    this.setState({ selectedPayment: value })
  }

  handleSubmit = () => {
    console.log('submitted')
  }

  render () {
    const { selectedPayment } = this.state
    return (
      <ScrollView>
        <View style={styles.container}>
          <StatusBar barStyle='light-content' />
          <Product />
          <Text style={styles.radioFormTitle}>
            Payment Method
          </Text>
          <PaymentTypes
            selectedPayment={selectedPayment}
            options={options}
            handleSelectChange={this.handleSelectChange}
          />
          <ContinueButton handleSubmit={this.handleSubmit} />
        </View>
      </ScrollView>
    )
  }
}

export default Checkout
