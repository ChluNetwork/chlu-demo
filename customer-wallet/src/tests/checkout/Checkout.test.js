import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Checkout from 'containers/Checkout'

describe('<Checkout />', () => {

  it('should render correctly checkout container', () => {
    const wrapper = shallow(
      <Checkout />
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
