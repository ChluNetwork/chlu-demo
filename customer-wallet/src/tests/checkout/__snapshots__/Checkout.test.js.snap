// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkout /> should render correctly checkout container 1`] = `
<div
  className="page-container checkout"
>
  <h3>
    Checkout at Etsy
  </h3>
  <div
    className="checkout-seller"
  >
    <Connect(SellerHeader) />
    <span
      className="checkout-seller__header-info"
    >
      (250 Reviews, Over $250,000 sales)
    </span>
    <Product />
    <Product />
  </div>
</div>
`;
